src/coordsim/writer/writer.py

```diff
@@ -5,33 +5,120 @@
 import csv
 import os
 import datetime as dt
-from spinterface import SimulatorAction
+from spinterface import SimulatorAction, SimulatorState
 
 
-def create_csv_stream():
+class ResultWriter():
     """
-    Creates a CSV file in append mode. Returns the opened stream
+    Result Writer module
     """
-    now = dt.datetime.now()
-    file_name = f"results/placements-{now.strftime('%d-%m-%Y--%H-%M-%S')}.csv"
-    os.makedirs(os.path.dirname(file_name), exist_ok=True)
-    stream = open(file_name, 'a+')
-    output_header = ['time', 'node', 'sf']
-    writer = csv.writer(stream)
-    writer.writerow(output_header)
-    return stream
+    def __init__(self, test_mode: bool):
+        """
+        If the simulator is in test mode, create result folder and CSV files
+        """
+        self.test_mode = test_mode
+        if self.test_mode:
+            now = dt.datetime.now()
 
+            self.scheduling_file_name = f"results/scheduling-{now.strftime('%d-%m-%Y--%H-%M-%S')}.csv"
```

Comment:
Could you use the same structure like in rl-coordination?
Then we have all the results of a test at a single directory. The current structure leads to a single folder with all result csv files.
Therefore we have to pass another value to the simulator.

Reply:
Sure. We can modify it to pass the folder to put the results in when calling the simulator interface. We will discuss this on slack